### PR TITLE
Rooks and King

### PR DESCRIPTION
**Pull request by ImMPrada**
This PR:
- upgrades Rook class to the correct namespace
- adds #can_castling and #castle to Chess::Core::Pieces::Rook, it  evaluates if a rook can is available for castling
- upgrades Chess::Core::Pieces::Rook specs
- upgrades King class to the correct namespace
- adds #can_castling and #castle to Chess::Core::Pieces::King, it  evaluates if a king can is available for castling and run the castling with a rook
- upgrades Chess::Core::Pieces::King specs

### CODE REVIEW

File: lib/chess/core/pieces/rook.rb [hunk 1]
```diff
@@ -0,0 +1,67 @@
+require_relative 'base_piece'
+
+module Chess
+  module Core
+    module Pieces
+      class Rook < BasePiece
+        SYMBOL = :R
+        TEXT = "\u265f".freeze
```
Comment (mapra99) [inline lib/chess/core/pieces/rook.rb:8]: is this needed?

File: lib/chess/core/pieces/rook.rb [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+          update_current_cell_to(target_cell)
+        end
+
+        def queen_side?
+          side == QUEEN_SIDE
+        end
+
+        def king_side?
+          side == KING_SIDE
```
Comment (mapra99) [inline lib/chess/core/pieces/rook.rb:34]: where are these constants defined?

File: spec/lib/chess/core/pieces/king_spec.rb [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+      it 'returns nil if rook was moved' do
+        rook.move_to(cell_c1, cells)
+        expect(king.castle_with(rook, cells)).to be_nil
+      end
+
+      it 'returns nil if king was moved' do
+        king.move_to(cell_f1, cells)
+        king.move_to(cell_e1, cells)
+        expect(king.castle_with(rook, cells)).to be_nil
+      end
```
Comment (mapra99) [inline spec/lib/chess/core/pieces/king_spec.rb:111]: remember what we talked about my preference towards not having anything other than `expect` inside examples? For these scenarios I suggest this:

```suggestion
      describe 'when rook was moved' do
        before do
          rook.move_to(cell_c1, cells)
        end
        
        it 'returns nil' do
          expect(king.castle_with(rook, cells)).to be_nil
        end
      end
      
      describe 'when king was moved' do
        before do
          king.move_to(cell_f1, cells)
          king.move_to(cell_e1, cells)
        end
        
        it 'returns nil' do
          expect(king.castle_with(rook, cells)).to be_nil
        end
      end
```

It's just a suggestion anyways

File: spec/lib/chess/core/pieces/king_spec.rb [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+        expect(cell_f1.occupant).to be(rook)
+      end
+
+      it 'occups the right cell with the king' do
+        king.castle_with(rook, cells)
+        expect(cell_g1.occupant).to be(king)
+      end
+    end
+  end
+  # rubocop:enable RSpec/MultipleMemoizedHelpers
```
Comment (mapra99) [inline spec/lib/chess/core/pieces/king_spec.rb:165]: I'm seeing you are violating this rule pretty often. Why don't you modify the .rubocop.yml file to increase the threshold?

**Review by mapra99 — APPROVED** [review]
GO GO GO!

File: spec/lib/chess/core/pieces/king_spec.rb (hunk 4 above)
Comment (ImMPrada) [reply spec/lib/chess/core/pieces/king_spec.rb:165]: It makes sense; it's possible I'll make that offense again Jajaja

**Pull request merged**